google/cloud/storage/internal/curl_upload_request.cc

```diff
@@ -28,9 +28,15 @@ namespace internal {
 
 CurlUploadRequest::CurlUploadRequest()
     : headers_(nullptr, &curl_slist_free_all),
-      multi_(nullptr, &curl_multi_cleanup) {}
+      multi_(nullptr, &curl_multi_cleanup),
+      closing_(false),
+      curl_closed_(false) {
+  buffer_.reserve(128 * 1024);
```

Comment:
Why 128K?

Reply:
No particular reason, and also a bad a idea. Made it into a parameter configurable in the builder.